framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -282,7 +282,7 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		 */
 		const lastSentCCM = (await this._chainConnectorStore.getLastSentCCM()) ?? {
 			nonce: BigInt(0),
-			height: this._lastCertificate.height,
+			height: this._lastCertificate?.height,
```

Comment:
is this really throwing the error? coz at this point in the code lastCertificate should exist

Reply:
we check for lastCertificate on L209 :) but I think we can still keep this change if its throwing warning or something

Reply:
i think it shouldn't throw error or warning because it's defined as
```
private _lastCertificate!: LastCertificate;
```
the code wise it looks strange to allow `height: undefined`

Reply:
<img width="1489" alt="image" src="https://user-images.githubusercontent.com/10962699/222776768-3631fc29-6ad3-4ac1-9b37-e41edb82e9d2.png">


error on this line

Reply:
Ahh even better solution 🙂

Reply:
Hi @shuse2 , I've updated the package.json scripts but getting the same failed message. Is there any other test setup that's off that I should have a look at? The `height` is still coming in as `undefined`.

Reply:
not only the package.json, but you need to fix the test. (I think in this case, test setup is wrong)